java/tests/com/google/privacy/differentialprivacy/CountTest.java

```diff
@@ -512,4 +513,160 @@ private static void testForBias(
+            .isEqualTo(ConfidenceInterval.create(0, 10));
+  }
+
+  @Test
+  public void infiniteUpperBound_roundsToMaxLong() {
+    when(noise.computeConfidenceInterval(
+            anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(1, Double.POSITIVE_INFINITY));
+    count.computeResult();
+
+    // The result interval = (1, POSITIVE_INFINITY), but because of the long type, it will be (1, Long.MAX_VALUE).
+    assertThat(count.computeConfidenceInterval(ALPHA))
+            .isEqualTo(ConfidenceInterval.create(1, Long.MAX_VALUE));
+  }
+
+  @Test
+  public void computeConfidenceInterval_forGaussianNoise() {
+    // Mock the noise mechanism.
+    when(noise.computeConfidenceInterval(anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new GaussianNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], (Double) invocation.getArguments()[4], (Double) invocation.getArguments()[5]));
+    count =
+            Count.builder()
+                    .epsilon(EPSILON)
+                    .delta(DELTA)
+                    .noise(noise)
+                    .maxPartitionsContributed(15)
+                    .build();
+    count.increment();
+    count.computeResult();
+
+    // The result interval = (-12, 14), but count can't be negative, therefore it will be clamped to 0.
+    assertThat(count.computeConfidenceInterval(ALPHA)).isEqualTo(ConfidenceInterval.create(0, 14));
+  }
+
+  @Test
+  public void computeConfidenceInterval_forLaplaceNoise() {
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be
+    // passed as a value instead of null, in order to pass the checks.
+    when(noise.computeConfidenceInterval(anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], null, (Double) invocation.getArguments()[5]));
+    count =
+            Count.builder()
+                    .epsilon(EPSILON)
+                    .noise(noise)
+                    .delta(DELTA)
+                    .maxPartitionsContributed(12)
+                    .build();
+    count.incrementBy(10);
+    count.computeResult();
+
+    // The result interval = (-58, 78), but count can't be negative, therefore it will be clamped to 0.
+    assertThat(count.computeConfidenceInterval(0.5)).isEqualTo(ConfidenceInterval.create(0, 78));
+  }
+
+  @Test
+  public void computeConfidenceInterval_whenComputeResultWasNotCalled_throwsException() {
```

Comment:
just computeResultWasNotCalled, without "when" for consistency with other tests.